### PR TITLE
create NavGraph and first implementation

### PR DESCRIPTION
**Pull request by A1ishka**

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -29,18 +31,21 @@ android {
...
-    }
-    kotlinOptions {
-        jvmTarget = "1.8"
+//    compileOptions {
+//        sourceCompatibility = JavaVersion.VERSION_1_8
+//        targetCompatibility = JavaVersion.VERSION_1_8
+//    }
+//    kotlinOptions {
+//        jvmTarget = "1.8"
+//    }
```
Comment (MaryFilipovichWork) [inline app/build.gradle.kts]: don't leave code in commets

File: app/src/main/java/com/a1ishka/bankaccount/navigation/NavGraph.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+        composable(
+            route = Screen.TransactionDetailsScreen.route + "?id=${id}",
+            arguments = listOf(
+                navArgument("id") {
+                    type = NavType.LongType
+                }
+            )
+        ) { entry ->
+//            TransactionDetailsScreen(id = entry.arguments?.getLong("id"))
```
Comment (MaryFilipovichWork) [inline app/src/main/java/com/a1ishka/bankaccount/navigation/NavGraph.kt]: don't leave code in comments

File: app/src/main/java/com/a1ishka/bankaccount/screens/accountDashboard/AccountDashboardScreen.kt [hunk 3]
```diff
@@ -67,8 +70,8 @@ fun AccountDashboard() {
                 .padding(top = 50.dp)
         ) {
             Column {
-                AccountCard(account = accountData[2])
-                RecentTransactionTitle()
+                AccountCard(account = accountData[2], onNavigate = { navController.navigate(Screen.AccountAddingScreen.route) })
+                RecentTransactionTitle(onNavigate = { navController.navigate(Screen.AllTransactionScreen.route) })
```
Comment (MaryFilipovichWork) [inline app/src/main/java/com/a1ishka/bankaccount/screens/accountDashboard/AccountDashboardScreen.kt]: it's easier to read the code when set parameters on separate lines (just recommendation)

File: app/src/main/java/com/a1ishka/bankaccount/navigation/NavGraph.kt [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        }
+
+        composable(route = Screen.AllTransactionScreen.route) {
+            AllTransaction()
+        }
+
+        composable(
+            route = Screen.TransactionDetailsScreen.route + "?id=${id}",
+            arguments = listOf(
+                navArgument("id") {
```
Comment (MaryFilipovichWork) [inline app/src/main/java/com/a1ishka/bankaccount/navigation/NavGraph.kt]: set names of arguments to constants to avoid mistakes

Comment (A1ishka) [reply app/src/main/java/com/a1ishka/bankaccount/navigation/NavGraph.kt]: added 

```
package com.a1ishka.bankaccount.util

object Constants {
    const val ARGUMENT_ID = "id"
}

File: app/src/main/java/com/a1ishka/bankaccount/screens/accountDashboard/AccountDashboardScreen.kt (hunk 3 above)
Comment (A1ishka) [reply app/src/main/java/com/a1ishka/bankaccount/screens/accountDashboard/AccountDashboardScreen.kt]: yes, earlier there were simple args, now moved into separates

```            
                AccountCard(
                    account = accountData[2],
                    onNavigate = {
                        navController.navigate(Screen.AccountAddingScreen.route)
                    }
                )
                RecentTransactionTitle(
                    onNavigate = {
                        navController.navigate(Screen.AllTransactionScreen.route)
                    }
                )

File: app/build.gradle.kts (hunk 1 above)
Comment (A1ishka) [reply app/build.gradle.kts]: deleted

File: app/src/main/java/com/a1ishka/bankaccount/navigation/NavGraph.kt (hunk 2 above)
Comment (A1ishka) [reply app/src/main/java/com/a1ishka/bankaccount/navigation/NavGraph.kt]: deleted

**Review by MaryFilipovichWork — APPROVED** [review]
approved

**Pull request merged**